conans/client/remover.py

```diff
@@ -75,65 +75,89 @@ def __init__(self, client_cache, search_manager, user_io, remote_proxy):
+        if has_remote and (build_ids is not None or src):
+            raise ConanException("Remotes don't have 'build' or 'src' folder, just packages")
+
+        conan_refs, package_ids = self._select_packages_to_remove(package_ids_filter, has_remote,
+                                                                  packages_query, pattern)
+
+        if not conan_refs:
+            if packages_query:
+                self._user_io.out.warn("No packages matching the query: %s" % packages_query)
+            else:
+                self._user_io.out.warn("No package recipe reference matches with '%s' pattern" % str(pattern))
             return
 
         deleted_refs = []
-        for conan_ref in search_info:
+        for conan_ref in conan_refs:
             assert(isinstance(conan_ref, ConanFileReference))
-            if self._ask_permission(conan_ref, src, build_ids, package_ids_filter, force):
+            if self._ask_permission(conan_ref, src, build_ids, package_ids, force):
                 if has_remote:
-                    if package_ids_filter is None:
+                    if package_ids is None:
                         self._remote_proxy.remove(conan_ref)
                     else:
-                        self._remote_proxy.remove_packages(conan_ref, package_ids_filter)
+                        self._remote_proxy.remove_packages(conan_ref, package_ids)
                 else:
                     deleted_refs.append(conan_ref)
                     remover = DiskRemover(self._client_cache)
                     if src:
                         remover.remove_src(conan_ref)
                     if build_ids is not None:
                         remover.remove_builds(conan_ref, build_ids)
-                    if package_ids_filter is not None:
-                        remover.remove_packages(conan_ref, package_ids_filter)
-                    if not src and build_ids is None and package_ids_filter is None:
+                    if package_ids is not None:
+                        remover.remove_packages(conan_ref, package_ids)
+                    if not src and build_ids is None and package_ids is None:
                         remover.remove(conan_ref)
                         registry = self._remote_proxy.registry
                         registry.remove_ref(conan_ref, quiet=True)
 
         if not has_remote:
             self._client_cache.delete_empty_dirs(deleted_refs)
 
+    def _select_packages_to_remove(self, package_ids, has_remote, packages_query, pattern):
+        if has_remote:
+            if not packages_query:
+                conan_refs = self._remote_proxy.search(pattern)
+            else:
+                package_ids = self._remote_proxy.search_packages(pattern, packages_query).keys()
+                conan_refs = [pattern] if package_ids else None
+        else:
+            if not packages_query:
+                conan_refs = self._search_manager.search(str(pattern))
```

Comment:
``packages_ids`` input parameter, not used for logic, just returned? Sounds weird